### PR TITLE
[chore] Add anchors to individual attributes in a markdown table #1041

### LINKED ISSUE #1041: Add anchors to individual attributes in a markdown table

### PR DESCRIPTION
**Pull request by RichardChukwu**
Fixes #1041 

## Changes

- Added anchor tags to individual attributes in the markdown table.

- Each attribute now has a corresponding <a id="attribute-name"></a> element, allowing direct linking to the specific attribute from the table of contents or other parts of the documentation.

- The anchor is dynamically generated using the attribute’s name, converted to a URL-friendly format (lowercase, spaces replaced with hyphens).

- This ensures that each attribute can be easily referenced by creating links to the anchor IDs.

## Merge requirement checklist

* [ ] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [ ] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: docs/attributes-registry/android.md
Comment (joaopgrassi) [inline docs/attributes-registry/android.md:19]: I believe this is not a good option, as it makes the pages long. 

I tried and I think it should work if we use simple HTML anchors. For example

```
<a id="android-os.api_level" href="#android-os.api_level">`android.os.api_level`</a>
```

Comment (joaopgrassi) [reply docs/attributes-registry/android.md:19]: This should make both the cell text clickable and add a anchor to the id defined to the URL.

**Review by joaopgrassi — CHANGES_REQUESTED** [review]

File: docs/attributes-registry/android.md
Comment (RichardChukwu) [reply docs/attributes-registry/android.md:19]: yes @joaopgrassi I looked into that and noticed the difference too. Let me use the HTML anchors and see, or are you working on a commit already?

Comment (RichardChukwu) [reply docs/attributes-registry/android.md]: Thanks for the feedback @joaopgrassi just tired it and it worked while retaining the table's structure.

Kindly review recent change and let me know how it looks

File: docs/attributes-registry/webengine.md
Comment (joaopgrassi) [reply docs/attributes-registry/webengine.md]: @RichardChukwu this was just an example, you need to fix it in the Jinja template to fix it in all files.

Comment (RichardChukwu) [reply docs/attributes-registry/webengine.md]: Ooh, on it please

File: templates/registry/markdown/attribute_namespace.md.j2
Comment (RichardChukwu) [reply templates/registry/markdown/attribute_namespace.md.j2:42]: Do you mean the attribute_namespace.md.j2 file instead?, because I did not edit the attribute_macro.j2 at all

Comment (RichardChukwu) [reply templates/registry/markdown/attribute_namespace.md.j2:42]: Understood now, working on it....

Comment (RichardChukwu) [reply templates/registry/markdown/attribute_namespace.md.j2:42]: > You need to remove the usage of the `attrs.name` function for the id and href. That adds the backticks around the attribute name. Please see it in attribute_macros.j2.

Tried running the jinja template using this 

> back ticks should not be added to id/hrefs. Not sure why they are being added, please check the jinja template.

The generated files seem distorted and anchor links removed too, unlike the previous solution:

image

Comment (RichardChukwu) [reply templates/registry/markdown/attribute_namespace.md.j2:42]: Ooh, thank you.

So in essence, reference the attribute.id field directly for the anchor links, while still using attrs.name for display purposes.

Let me try that out again

Comment (RichardChukwu) [reply templates/registry/markdown/attribute_namespace.md.j2]: Kindly check recent push please if all is modified

File: templates/registry/markdown/attribute_namespace.md.j2
Comment (joaopgrassi) [inline templates/registry/markdown/attribute_namespace.md.j2:42]: Again, if you look at attribute_macros.j2 you will see that it uses `attribute.name`. There's no `id`. Now, all id/href are empty...

Comment (RichardChukwu) [reply templates/registry/markdown/attribute_namespace.md.j2]: How about now? @joaopgrassi  I apologize for the back and forth, I seemed not to be getting it before.

**Review by joaopgrassi — APPROVED** [review]

**Comment by joaopgrassi** [comment]
Ah.. I forgot that the link check may not work/like the anchor links in the URL 😞

**Comment by RichardChukwu** [comment]
Ooh, alright

**Comment by joaopgrassi** [comment]
I pushed a workaround that skips local anchor links for now. Let's see if it passes.

**Comment by RichardChukwu** [comment]
Hey @joaopgrassi I guess the test passed?

Thanks for the workaround

**Comment by joaopgrassi** [comment]
Needs more approvals, so we need to wait for others to review it as well.

**Review by trisch-me — APPROVED** [review]
thanks! lgtm

**Review by lmolkova — APPROVED** [review]
Thanks for doing this!

It would also be great to update the markdown tables that already reference registry attributes. 

E.g. here 
https://github.com/open-telemetry/semantic-conventions/blob/e038c11b8773ca53045ad490e2ea046a7ab01e24/docs/http/http-spans.md?plain=1#L145

**Comment by RichardChukwu** [comment]
@lmolkova Sure, but I suggest that can be addressed in a different PR as a different issue for clarity sake.

Your thoughts?

**Comment by lmolkova** [comment]
@RichardChukwu sure!

**Pull request merged**
